### PR TITLE
TypeScript support

### PR DESCRIPTION
**Pull request by pittst3r**
Required config on the ember side: https://github.com/emberjs/ember.js/pull/14896

### CODE REVIEW

**Comment by robbiepitts** [comment]
Need to tweak this to pass in node_modules to the Typescript compiler.

File: lib/get-es6-package.js [hunk 1]
```diff
@@ -46,6 +47,10 @@ function getES6Package(packages, packageName, opts) {
   */
   var files = [ jsRegExp ];
 
+  if (pkg.hasTypescript) {
```
Comment (ZeeJab) [inline lib/get-es6-package.js]: This should be camel cased (`hasTypeScript`)

Comment (tomdale) [inline lib/get-es6-package.js]: This should be camel cased (`isTypeScript`).

**Comment by robbiepitts** [comment]
This requires a package-specific tsconfig.json like:

```
$ more packages/container/tsconfig.json
{
  "compilerOptions": {
    "target": "es2015",
    "module": "es2015",
    "moduleResolution": "node",
    "baseUrl": ".",
    "paths": {
      "*": ["packages/*/lib/index"]
    }
  },
  "include": [
    "container/**/*.ts"
  ],
  "exclude": [
    "node_modules",
    "packages",
    "**/*.d.ts"
  ]
}
```

**Comment by robbiepitts** [comment]
What do we want to do with the declaration files that will be generated?

**Comment by chadhietala** [comment]
Has this been tested in VSCode with some typescript files to see if the type definitions are followed and TypeScript language service doesn't become over loaded? /cc @krisselden

**Comment by rwjblue** [comment]
@chadhietala - Not to my knowledge.  I'm really unsure how to use VSCode for this. The main purpose here is to get things functioning so that work integrating `@glimmer/di` can begin.

**Comment by robbiepitts** [comment]
@chadhietala If you're asking if putting TS in emberjs in VSCode has been tested, yes. Type definitions are followed and I have not noticed any problems. Rebuilds while running tests in the ember repo are also working fine.

glimmer-container

**Comment by rwjblue** [comment]
@robbiepitts - Is this still WIP, if yes, can you add a TODO checklist to the description?

**Comment by robbiepitts** [comment]
@rwjblue Sorry, no, this is good to go. I made it WIP when I ran into some issues that turned out to be tsconfig related.

I'll update the PR description to include config details.

**Comment by rwjblue** [comment]
I'm sorry we didn't get back to this previously, but at this point this repo is unused by Ember and unmaintained.

Closing...

**Pull request closed without merging**